tests/framework/resourcemanager.go

```diff
@@ -719,6 +719,40 @@ func GetReadyNGFPodNames(
 	return nil, errors.New("unable to find NGF Pod(s)")
 }
 
+// GetReadyNginxPodNames returns the name(s) of the Nginx Pod(s).
+func GetReadyNginxPodNames(
+	k8sClient client.Client,
+	namespace string,
+	timeout time.Duration,
+) ([]string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	var podList core.PodList
+	if err := k8sClient.List(
+		ctx,
+		&podList,
+		client.InNamespace(namespace),
+		client.HasLabels{"gateway.networking.k8s.io/gateway-name"},
+	); err != nil {
+		return nil, fmt.Errorf("error getting list of Nginx Pods: %w", err)
+	}
+
+	if len(podList.Items) > 0 {
+		var names []string
+		for _, pod := range podList.Items {
+			for _, cond := range pod.Status.Conditions {
+				if cond.Type == core.PodReady && cond.Status == core.ConditionTrue {
+					names = append(names, pod.Name)
+				}
+			}
+		}
+		return names, nil
+	}
+
+	return nil, errors.New("unable to find NGF Pod(s)")
+}
```

Comment:
nit: check if podList.Items is empty, and if so, return the error. Then you can just have the loop outside that conditional after that check.